### PR TITLE
Fix Regex setting in case of valueIsRegex

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
Prevent updating the regex if valueIsRegex is true.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?
Added tests in ingest.csv and restart.csv

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Review by nkunal — COMMENTED** [review]
lgtm

File: pkg/ast/common.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func ProcessSingleFilter(colName string, colValue interface{}, originalColValue
 						log.Errorf("qid=%d, ProcessSingleFilter: Failed to compile regex for %s. This may cause search failures. Err: %v", qid, t, err)
 						return nil, fmt.Errorf("invalid regex: %s", t)
 					}
-					criteria := CreateTermFilterCriteria(colName, compiledRegex, opr, qid, caseConversion)
+					criteria := CreateTermFilterCriteria(colName, compiledRegex, opr, qid, caseConversion, valueIsRegex)
```
Comment (Macbeth98) [inline pkg/ast/common.go:107]: `valueIsRegex` is in `caseConversion` struct. You do not need to pass it as a param here.

Same for other places.

File: pkg/ast/common.go [hunk 2]
```diff
@@ -258,19 +258,19 @@ func createMatchFilterCriteria(colName, colValue interface{}, opr LogicalOperato
...
 
-func CreateTermFilterCriteria(colName string, colValue interface{}, opr FilterOperator, qid uint64, cci *CaseConversionInfo) *FilterCriteria {
+func CreateTermFilterCriteria(colName string, colValue interface{}, opr FilterOperator, qid uint64, cci *CaseConversionInfo, valueIsRegex bool) *FilterCriteria {
 	cVal, err := CreateDtypeEnclosure(colValue, qid)
 	if err != nil {
 		log.Errorf("qid=%d, createTermFilterCriteria: error creating DtypeEnclosure for ColValue=%v. Error=%+v", qid, colValue, err)
 	} else {
-		if cci != nil {
+		if cci != nil && !valueIsRegex {
 			cVal.UpdateRegexp(cci.caseInsensitive)
```
Comment (Macbeth98) [inline pkg/ast/common.go:267]: Move this inside the below if condition. We only need to do this at that place.

Comment (Macbeth98) [reply pkg/ast/common.go:267]: `ShouldAlsoSearchWithOriginalCase` will ensure that if `valueIsRegex` set to true, it won't be considered as a case-insensitive search.

File: pkg/ast/common.go (hunk 1 above)
Comment (vasukalariya) [reply pkg/ast/common.go]: addressed in the latest commit.

File: pkg/ast/common.go (hunk 2 above)
Comment (vasukalariya) [reply pkg/ast/common.go:267]: addressed in the latest commit.

**Review by Macbeth98 — APPROVED** [review]

**Pull request merged**